Sources/Corvus/Endpoints/SecureCRUD.swift

```diff
@@ -0,0 +1,94 @@
+import Vapor
+import Fluent
+
+/// A class that contains Create, Read, Update and Delete functionality for a
+/// generic type `T` conforming to `CorvusModel` grouped under a given path and
+/// allows securing those resources with the `.auth()` modifier.
+public final class SecureCRUD<
+    T: CorvusModel,
+    A: CorvusModelAuthenticatable
+>: Endpoint {
+
+    /// The route path to the parameters.
+    let pathComponents: [PathComponent]
+
+    /// A property to generate route parameter placeholders.
+    let parameter = Parameter<T>()
+
+    /// Indicates whether soft delete should be included or not.
+    let useSoftDelete: Bool
+    
+    /// The `KeyPath` to the user property of the `QuerySubject` which is to be
+    /// authenticated.
+    public typealias UserKeyPath = KeyPath<
+        T,
+        T.Parent<A>
+    >
+    
+    /// The path to the property to authenticate for.
+    public var userKeyPath: UserKeyPath
+    
+    /// Initializes the component with one or more route path components.
+    ///
+    /// - Parameter pathComponents: One or more `PathComponents` identifying the
+    /// path to the operations defined by the `SecureCRUD` component.
+    /// - Parameter softDelete: Enable/Disable soft deletion of Models.
+    /// - Parameter userKeyPath: The path to the user property of the model `T`
+    /// the component operates on.
+    public init(
+        _ pathComponents: PathComponent...,
+        user userKeyPath: UserKeyPath,
+        softDelete: Bool = true
```

Comment:
I would suggest that CRUD as well as SecureCRUD have a default behavior `softDelete: Bool = true` false as a soft delete is not the default behavior of `Delete` and CRUD implies Create, Read, Update and Delete and not CRUSD 👍
I think that soft delete should be an explicit opt in of the developer.

On a related note, we could simplify the API surface a bit if we would move the soft delete functionality into the Delete Endpoint and expose it as a init parameter the same way we do it here.

Reply:
Both done